torchdata/dataloader2/README.md

```diff
@@ -0,0 +1,68 @@
+# DataLoader2 (Prototype)
+
+`DataLoader2` is introduced to execute `DataPipe` graph, with support for
+
+- dynamic sharding for multi-process and distributed data loading
+- multiple backend `ReadingServices` (WIP for `DistributedReadingService`)
+- `DataPipe` graph in-place modification like shuffle control, memory pinning, etc.
+- snapshot the state of data-preprocessing pipeline (WIP)
+
+## Why DataLoader2
+
+`DataPipe` is introduced to not only decompose the data pre-processing operations, but also decouple the overloaded
+data-manipulation features from `DataLoader`. So, a light-weight `DataLoader2` is required. Besides, a certain features
+can only be achieved with `DataLoader2` like snapshotting and bridging data-processing pipeline between research and
```

Comment:
```suggestion
data-manipulation features from `DataLoader`. So, a light-weight `DataLoader2` is required. Besides, certain features
can only be achieved with `DataLoader2`, such as snapshotting and bridging data-processing pipeline between research and
```

Reply:
I am actually not sure what feature you are referring to by "bridging data-processing pipeline between research and". What feature is that one?

Reply:
I think one feature you have in mind is `ReadingService`? If so we can directly mention it here.

Reply:
Let me rephrase it.